### DISCUSSION TITLE
Get taxonomy on collection's detail page?

### DISCUSSION THREAD

**Discussion by torbentschechne** [statamic/cms#5298] [Q&A]
Hi there,
I have a collection "pages", within this collection I have an entry like "Installation Guide". This entry has a taxonomy "version" with the value "version5".

Now when I am on the detail page of this pages entry in the frontend I just want to check if the version's value is "version5" or "version6". So I am looking for something like `{{ has_version('version5') }}` or `{{ has_taxonomy_value('version', 'version5') }}` or `{{ taxonomy:version.title }}` to get the value. I also need to pass this value to antother route to filter the items. Is there a function for this?

**Comment by ryanmitchell** [comment]
If you do a {{ dump }} on the 'show' page you should see the variables available, one of which should be your taxonomy field. You can then loop over it {{ your_field_name }} .... {{ /your_field_name }} to see the terms attached to it.

**Comment by torbentschechne** [comment]
@ryanmitchell Yes, that works. But now I want to check if a certain page has two different terms. So I have a taxonomy `version` and a taxonomy `tags`. So what I basically need is something like:

`{if version|contains:"sw5" && tags|contains:"newsletter"}`

But that does not exists, right? Do I need to write my own tag here to check the terms of a page?

**Reply by ryanmitchell** [reply]
I havent tested but can you do

`{{ if (version|contains:"sw5") and (tags|contains:"newsletter") }}`

?

**Reply by torbentschechne** [reply]
Unfortunately not - that gives me syntax errors and when I do this:

`{{ if {version|contains:"sw5} and {tags|contains:"newsletter"} }}`

With curly brackets I do not have syntax errors, but no result either...

**Reply by ryanmitchell** [reply]
Hmm ok, maybe a custom tag would be easier. I've a feeling what I suggested will work in the new parser in 3.3

**Reply by torbentschechne** [reply]
I looked into the custom tags recently, but what I do not see is: How can I retrieve the current item where the tag is called? So when I have the current item, I must retrieve the taxonomy's which are assigned to the item. Do you have a hint?

**Reply by ryanmitchell** [reply]
You would do something like {{ my_tag :entry="entry_variable" }} and then within your tag you can get the entry using $this->params('entry')

**Reply by torbentschechne** [reply]
But I can not get the whole entry object through a variable, right? So what I see in the dump on my collection item is an entry_id that I can pass, but not the whole object. With the id I need to get this item again via the repository, right?

**Reply by ryanmitchell** [reply]
Yeah or you could just pass the two taxonomy variables?

{{ my_tag :version ="version" :tags="tags" }}

**Reply by torbentschechne** [reply]
Yep, I did that already locally. I think I can solve it that way. Thanks for your help along the way!
